packages/lookup/src/ux-lookup.ts

```diff
@@ -0,0 +1,329 @@
+  }
+
+  debounceNumber: number | null | undefined = this.defaultConfiguration.debounce;
+  @bindable
+  debounce: number | string | undefined = this.defaultConfiguration.debounce;
+  debounceChanged() {
+    this.debounceNumber = normalizeNumberAttribute(this.debounce);
+  }
+
+  bind() {
+    this.themeChanged(this.theme);
+    this.valueFieldChanged();
+    this.displayFieldChanged();
+    this.optionsChanged();
+  }
+
+  attached() {
+    this.inputElement = this.element.parentElement?.querySelector<UxInputElement>('ux-input');
+    if (this.inputElement) {
+      ['click', 'blur', 'change', 'keydown'].map(x => this.inputElement!.addEventListener(x, this));
+    }
+    ['blur', 'keydown'].map(x => this.element.addEventListener(x, this));
+    this.valueChanged();
+  }
+
+  detached() {
+    if (this.inputElement) {
+      ['click', 'blur', 'change', 'keydown'].map(x => this.inputElement!.removeEventListener(x, this));
+    }
+    ['blur', 'keydown'].map(x => this.element.removeEventListener(x, this));
+  }
+
+  async open() {
+    if (this.isOpen) {
+      return;
+    }
+    this.updateAnchor();
+    ['wheel', 'scroll'].map(x => window.addEventListener(x, this, true));
+    this.taskQueue.queueTask(() => this.isOpen = true);
+  }
+
+  close() {
+    this.isOpen = false;
+    this.focusedOption = undefined;
+    ['wheel', 'scroll'].map(x => window.addEventListener(x, this, true));
+  }
+
+  updateAnchor() {
+    if (!this.inputElement) {
+      return;
+    }
+    const inputRect = this.inputElement.getBoundingClientRect();
+    const style = getComputedStyle(this.element);
+    const inputDistance = parseInt(style.getPropertyValue('--aurelia-ux--lookup-input-distance') ?? UxLookupTheme.DEFAULT_INPUT_DISTANCE.toString());
+    const windowEdgeDistance = parseInt(style.getPropertyValue('--aurelia-ux--lookup-window-edge-distance') ?? UxLookupTheme.DEFAULT_WINDOW_EDGE_DISTANCE.toString());
+    const bottomHeightThreshold = parseInt(style.getPropertyValue('--aurelia-ux--lookup-bottom-height-threshold') ?? UxLookupTheme.DEFAULT_BOTTOM_HEIGHT_THRESHOLD.toString());
+    let availableHeight = document.body.scrollTop + window.innerHeight - inputRect.bottom - inputDistance - windowEdgeDistance;
+    if (availableHeight > bottomHeightThreshold) {
+      this.anchor = {
+        left: inputRect.left,
+        top: `${inputRect.top + inputRect.height + inputDistance}px`,
+        bottom: undefined,
+        maxHeight: availableHeight,
+        width: inputRect.width
+      };
+    } else {
+      availableHeight = inputRect.top - document.body.scrollTop - inputDistance - windowEdgeDistance;
+      this.anchor = {
+        left: inputRect.left,
+        top: undefined,
+        bottom: `${window.innerHeight - availableHeight - windowEdgeDistance}px`,
+        maxHeight: availableHeight,
+        width: inputRect.width
+      };
+    }
+  }
+
+  handleEvent(evt: Event): void {
+    if (evt.currentTarget === this.inputElement) {
```

Comment:
it's a weird looking combination of if and switch, though I like how it's organised

Reply:
fixed